SafeExamBrowser.UserInterface.Desktop/Windows/BrowserWindow.xaml.cs

```diff
@@ -329,6 +424,47 @@ private void RegisterEvents()
 			ZoomInButton.Click += (o, args) => ZoomInRequested?.Invoke();
 			ZoomOutButton.Click += (o, args) => ZoomOutRequested?.Invoke();
 			ZoomResetButton.Click += (o, args) => ZoomResetRequested?.Invoke();
+			BrowserControlHost.GotKeyboardFocus += BrowserControlHost_GotKeyboardFocus;
+		}
+
+		private void MenuButton_Click(object sender, RoutedEventArgs e)
+		{
+			MenuPopup.IsOpen = !MenuPopup.IsOpen;
+			ZoomInButton.Focus();
+		}
+
+		private void BrowserControlHost_GotKeyboardFocus(object sender, KeyboardFocusChangedEventArgs e)
+		{
+			var forward = !this.browserControlGetsFocusFromTaskbar;
+
+			// focus the first / last element on the page
+			var javascript = @"
+if (typeof __SEB_focusElement === 'undefined') {
+  __SEB_focusElement = function (forward) {
+    var items = [].map
+      .call(document.body.querySelectorAll(['input', 'select', 'a[href]', 'textarea', 'button', '[tabindex]']), function(el, i) { return { el, i } })
+      .filter(function(e) { return e.el.tabIndex >= 0 && !e.el.disabled && e.el.offsetParent; })
+      .sort(function(a,b) { return a.el.tabIndex === b.el.tabIndex ? a.i - b.i : (a.el.tabIndex || 9E9) - (b.el.tabIndex || 9E9); })
+    var item = items[forward ? 1 : items.length - 1];
+    setTimeout(function () { item.focus(); }, 20);
+  }
+}";
+			var control = BrowserControlHost.Child as IBrowserControl;
+			control.ExecuteJavascript(javascript, result =>
+			{
+				if (!result.Success)
+				{
+					//logger.Error($"Javascript error {result.Message}!");
```

Comment:
Please remove all commented-out code if it is not needed.

Reply:
Done.

Reply:
Yes, if you think that the information could be useful to investigate issues then we should definitely log it (e.g. using `logger.Error(...)` or whichever severity you see fit). There isn't any internal logging at the moment, its all either in the application log or otherwise gone once the application terminates.